modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/UserResourceProfileRepository.java

```diff
@@ -0,0 +1,227 @@
+            logger.debug("Populating the Primary Key UserComputeResourcePreferences objects for the User Resource Profile");
+            userResourceProfileEntity.getUserComputeResourcePreferences().forEach(userComputeResourcePreferenceEntity -> userComputeResourcePreferenceEntity.setUserId(userId));
+            userResourceProfileEntity.getUserComputeResourcePreferences().forEach(userComputeResourcePreferenceEntity -> userComputeResourcePreferenceEntity.setGatewayId(gatewayId));
+        }
+
+        if (userResourceProfileEntity.getUserStoragePreferences() != null) {
+            logger.debug("Populating the Primary Key UserStoragePreferences objects for the User Resource Profile");
+            userResourceProfileEntity.getUserStoragePreferences().forEach(userStoragePreferenceEntity -> userStoragePreferenceEntity.setUserId(userId));
+            userResourceProfileEntity.getUserStoragePreferences().forEach(userStoragePreferenceEntity -> userStoragePreferenceEntity.setGatewayId(gatewayId));
+        }
+
+        if (!isUserResourceProfileExists(userId, gatewayId)) {
+            logger.debug("Checking if the User Resource Profile already exists");
+            userResourceProfileEntity.setCreationTime(new Timestamp(System.currentTimeMillis()));
+        }
+
+        userResourceProfileEntity.setUpdateTime(new Timestamp(System.currentTimeMillis()));
+        return execute(entityManager -> entityManager.merge(userResourceProfileEntity));
+    }
+
+    @Override
+    public String addUserResourceProfile(UserResourceProfile userResourceProfile) throws AppCatalogException {
+        return saveUserResourceProfileData(userResourceProfile);
+    }
+
+    @Override
+    public void updateUserResourceProfile(String userId, String gatewayId, UserResourceProfile updatedProfile) throws AppCatalogException {
+        saveUserResourceProfileData(updatedProfile);
+    }
+
+    @Override
+    public UserResourceProfile getUserResourceProfile(String userId, String gatewayId) throws AppCatalogException {
+        Map<String, Object> queryParameters = new HashMap<>();
+        queryParameters.put(DBConstants.UserResourceProfile.USER_ID, userId);
+        queryParameters.put(DBConstants.UserResourceProfile.GATEWAY_ID, gatewayId);
+        List<UserResourceProfile> userResourceProfileList = select(QueryConstants.GET_USER_RESOURCE_PROFILE, -1, 0, queryParameters);
+
+        if(!userResourceProfileList.isEmpty() && userResourceProfileList != null) {
+            logger.debug("Return the record (there is only one record)");
+            return userResourceProfileList.get(0);
+        }
+
+        return null;
+    }
+
+    @Override
+    public UserComputeResourcePreference getUserComputeResourcePreference(String userId, String gatewayId, String hostId) throws AppCatalogException {
+        UserComputeResourcePreferenceRepository userComputeResourcePreferenceRepository = new UserComputeResourcePreferenceRepository();
```

Comment:
Same comment here, should be able to do a `userComputeResourcePreferenceRepository.get()`.